framer/Animator.coffee

```diff
@@ -30,6 +30,82 @@ class exports.Animator
 				break
 		return values
 
+	@parseFunction: (str) ->
+		result = {name: "", args: []}
+
+		if _.endsWith str, ")"
+			result.name = str.split("(")[0]
+			result.args = str.split("(")[1].split(",").map (a) -> _.trim(_.trimEnd(a, ")"))
+		else
+			result.name = str
+
+		return result
+
+	@classForCurve: (curve) ->
+		parsedCurve = @parseFunction(curve)
+		animatorClassName = parsedCurve.name.toLowerCase()
+
+		if AnimatorClasses.hasOwnProperty(animatorClassName)
+			return AnimatorClasses[animatorClassName]
+
+		if animatorClassName in AnimatorClassBezierPresets
+			return BezierCurveAnimator
+
+		return null
+
+	@curveOptionsFor: (options={}) ->
+		curveOptions = options.curveOptions ? {}
+		animatorClass = @classForCurve(options.curve)
+		parsedCurve = @parseFunction options.curve
+		animatorClassName = parsedCurve.name.toLowerCase()
+
+		# This is for compatibility with the direct Animation.time argument. This should
+		# ideally also be passed as a curveOption
+
+		if animatorClass in [LinearAnimator, BezierCurveAnimator]
+			if _.isString(curveOptions) or _.isArray(curveOptions)
+				curveOptions =
+					values: curveOptions
+
+			curveOptions.time ?= options.time
+
+		# Support ease-in etc
+		if animatorClass in [BezierCurveAnimator] and animatorClassName in AnimatorClassBezierPresets
+			curveOptions.values = animatorClassName
+			curveOptions.time ?= options.time
+
+		# All this is to support curve: "spring(100, 20, 10)". In the future we'd like people
+		# to start using curveOptions: {tension:100, friction:10} etc
+
+		if parsedCurve.args.length
+
+			# console.warn "Animation.curve arguments are deprecated. Please use Animation.curveOptions"
+
+			if animatorClass is BezierCurveAnimator
+				curveOptions.values = parsedCurve.args.map (v) -> parseFloat(v) or 0
+
+			if animatorClass is SpringRK4Animator
+				for k, i in ["tension", "friction", "velocity", "tolerance"]
+					value = parseFloat parsedCurve.args[i]
+					curveOptions[k] = value if value
+
+			if animatorClass is SpringDHOAnimator
+				for k, i in ["stiffness", "damping", "mass", "tolerance"]
+					value = parseFloat parsedCurve.args[i]
+					curveOptions[k] = value if value
+		return curveOptions
 
 	# start: -> Framer.Loop.on("update", )
 	# stop: -> AnimationLoop.remove @
+
+{LinearAnimator} = require "./Animators/LinearAnimator"
+{BezierCurveAnimator} = require "./Animators/BezierCurveAnimator"
+{SpringRK4Animator} = require "./Animators/SpringRK4Animator"
+{SpringDHOAnimator} = require "./Animators/SpringDHOAnimator"
+AnimatorClasses["linear"] = LinearAnimator
+AnimatorClasses["bezier-curve"] = BezierCurveAnimator
+AnimatorClasses["spring-rk4"] = SpringRK4Animator
+AnimatorClasses["spring-dho"] = SpringDHOAnimator
```

Comment:
Yeah, but because of dependencies the AnimatorClasses object has to already exist, but the classes depend on the Animator Object, so I had to do it like this :/
Any suggestions how to improve this?